### PR TITLE
bug(race): Fixed races

### LINKED ISSUE #35: Tests should run cleanly when passed the -race option
To ensure our tests (and to some extent production code) are reliable, we should be running unit tests with the -race argument in the Makefile. Currently, the tests fail when run in this mode.

Failing tests: 
* TestRegisterWithPingCallback() in internal/pkg/consul/client_test.go
* TestIsServiceAvailableNotHealthy() in internal/pkg/consul/client_test.go
* TestConfigurationValueExists() in internal/pkg/consul/client_test.go
* TestPutConfiguration() in internal/pkg/consul/client_test.go
* TestWatchForChanges() in internal/pkg/consul/client_test.go

This test should succeed when passed -race, and then we should add -race to the test target in our Makefile.

### PR DESCRIPTION
**Pull request by brandonforster**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
We do not currently make use of the built in Go race detector.

Issue Number: Addressed as part of #35

## What is the new behavior?
We make use of the built in Go race detector.

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

### CODE REVIEW

**Comment by brandonforster** [comment]
@lenny-intel @rsdmike Any updates on this PR?

File: Makefile [hunk 1]
```diff
@@ -6,4 +6,4 @@ test:
 	$(GO) test ./... -coverprofile=coverage.out ./...
 	$(GO) vet ./...
 	gofmt -l .
-	[ "`gofmt -l .`" = "" ]
```
Comment (lenny-intel) [inline Makefile]: shouldn't -race be added to the `test` line?

Comment (brandonforster) [reply Makefile]: [Per Dev Ops WG call, no.](https://github.com/edgexfoundry/go-mod-registry/issues/35#issuecomment-592093762)

Comment (lenny-intel) [reply Makefile]: Ok, then. :-)

**Pull request merged**